app/routes/routes.py

```diff
@@ -0,0 +1,237 @@
+        }
+    return make_response(response_body, 201)
+
+# @task_bp.route("", methods=["GET"])
+# def read_all_task():
+#     tasks_response = []
+#     tasks = Task.query.all()
+#     for task in tasks:
+#         tasks_response.append(task.to_dict())
+#     return jsonify(tasks_response)
+##GET ALL TASKS AND SORT TASKS BY ASC & DESC
+
+@task_bp.route("", methods=["GET"])
+def read_all_task():
+    title_sort_query = request.args.get("sort")
+    if title_sort_query == "asc":
+        tasks = Task.query.order_by(Task.title.asc())
+    elif title_sort_query == "desc":
+        tasks = Task.query.order_by(Task.title.desc())
+    else:
+        tasks = Task.query.all()
+
+    response = []
+    for task in tasks:
+        response.append(task.to_dict())
+    return jsonify(response)
+
+#GET ONE TASK
+@task_bp.route("/<task_id>", methods=["GET"])
+def get_one_task(task_id):
+    task = validate_id(Task, task_id)
+    if task.goal_id is None:
+        return {"task": task.to_dict()}
+    else:
+        return {"task": task.to_new_dict()}
+    # response_body = {
+    #     "task": task.to_dict()
+    # }
+    # return response_body
+
+#UPDATE TASK
+@task_bp.route("/<task_id>", methods=["PUT"])
+def update_task(task_id):
+    task = validate_id(Task, task_id)
+    request_body = request.get_json() 
+    task.title = request_body["title"]
+    task.description = request_body["description"]
+    # TODO task.completed_at = request_body["completed_at"] #include later
+    # task.update(request_body)
+    db.session.commit()
+    response_body =  {
+        "task": task.to_dict()
+        }
+    return make_response(response_body, 200)
+
+#DELETE ONE TASK
+@task_bp.route("/<task_id>", methods=["DELETE"])
+def delete_task(task_id):
+    task = validate_id(Task, task_id)
+
+    task_dict = task.to_dict()
+
+    db.session.delete(task)
+    db.session.commit()
+
+    return {
+        "details": f'Task {task_id} "{task_dict["title"]}" successfully deleted'}
+
+#MARK COMPLETE
+#MODIFY MARK COMPLETE TO CALL SLACK API
+@task_bp.route("/<task_id>/mark_complete", methods=["PATCH"]) #custom endpoint mark task complete
+def mark_complete(task_id):
+    task = validate_id(Task, task_id)
+    task.completed_at = datetime.utcnow()
+
+    db.session.commit()
+    response = {
+        "task": task.to_dict()
+    }
+    slack_key = os.environ.get("SLACKBOT_API_KEY")
+    path = "https://slack.com/api/chat.postMessage"
+    data = {
+        "channel": "task-notifications",
+        "text": f"Someone just completed the task {task.title}"
+    }
+    headers = {
+        "authorization":f"Bearer {slack_key}"
+    }
+    return jsonify(response)
+
+
+#MARK INCOMPLETE
+@task_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def mark_incomplete(task_id):
+    task = validate_id(Task, task_id)
+    task.completed_at = None
+    
+    db.session.commit()
+    response = {
+        "task": task.to_dict()
+    }
+
+    return jsonify(response)
+
+### GOAL ROUTES
```

Comment:
I like the use of multiple `#`s in these headings. Gets the point across without being too fancy.